frontend/src/app/projects/models/backup-settings/view-logs-dialog/view-logs-dialog.component.ts

```diff
@@ -29,26 +29,31 @@ export class ViewLogsDialogComponent {
     this.refreshEvents();
   }
 
-  logs = '';
-  events = '';
+  logs?: string = undefined;
+  events?: string = undefined;
 
   refreshEvents(): void {
     combineLatest([
-      this.projectService.project$.pipe(filter(Boolean)),
-      this.modelService.model$.pipe(filter(Boolean)),
-      this.pipelineService.pipeline$.pipe(filter(Boolean)),
+      this.projectService.project$.pipe(filter(Boolean), take(1)),
+      this.modelService.model$.pipe(filter(Boolean), take(1)),
+      this.pipelineService.pipeline$.pipe(filter(Boolean), take(1)),
       this.pipelineRunService.pipelineRun$.pipe(filter(Boolean)),
     ])
       .pipe(
         untilDestroyed(this),
-        switchMap(([project, model, pipeline, pipelineRun]) =>
-          this.pipelineRunService.getEvents(
+        filter(
+          ([_project, _model, _pipeline, pipelineRun]) =>
+            this.logs === undefined ||
+            !this.pipelineRunService.pipelineRunIsFinished(pipelineRun.status)
+        ),
+        switchMap(([project, model, pipeline, pipelineRun]) => {
+          return this.pipelineRunService.getEvents(
             project.slug,
             model.slug,
             pipeline.id,
             pipelineRun.id
-          )
-        )
+          );
+        })
       )
       .subscribe({
         next: (res: string) => {
```

Comment:
```
next: (res: string) => (this.events = res),
error: () => (this.events = "Couldn't fetch events"),
```
Again nothing that you directly changed but for these one line assignments at least I prefer to just do it like this. Whats your opinion on that?